### PR TITLE
Added Filter by Category Tag.

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Filter by a category
- Users can filter videos by selecting any `category tag` on top of the video listing page. 
- by default, videos are filtered based on the `All` tag.

### CODE REVIEW

File: src/context/VideoContext.js [hunk 1]
```diff
@@ -4,6 +4,9 @@ import { videoReducer } from "../reducers";
 
 const initialState = {
   videos: [],
+  filters: {
+    category: "All",
+  },
   loading: false,
   error: null,
```
Comment (pritam-kr) [inline src/context/VideoContext.js:11]: I didn't understand why you give for every context, loading false and error.

File: src/helpers/helpers.js [hunk 2]
```diff
@@ -1,2 +1,17 @@
 export const getTrendingVideos = videos =>
   [...videos].sort((video1, video2) => video2.views - video1.views);
+
+export const getUniqueValues = (data, type) => {
+  let unique = data.map(item => item[type]);
+  return ["All", ...new Set(unique)];
+};
```
Comment (pritam-kr) [inline src/helpers/helpers.js:7]: 🔥

**Review by pritam-kr — APPROVED** [review]
- All categories working fine.
- clean code
- The UI is great.
- I added a point please explain this.

File: src/context/VideoContext.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/context/VideoContext.js:11]: It's just for initial fetching of data.

File: src/context/VideoContext.js [hunk 3]
```diff
@@ -14,6 +17,7 @@ const VideoProvider = ({ children }) => {
   const [state, dispatch] = useReducer(videoReducer, initialState);
 
   useEffect(() => {
+    dispatch({ type: "LOADING" });
```
Comment (kiranpratyush) [inline src/context/VideoContext.js:20]: I think this dispatch is for Video Reducer. But there is no  LOADING type declared in the reducer. Please correct me If I am missing something.

File: src/helpers/helpers.js [hunk 4]
```diff
@@ -1,2 +1,17 @@
...
+
+export const getFilteredVideos = (videos, state) => {
+  if (state.filters.category !== "All") {
+    return [...videos].filter(
```
Comment (kiranpratyush) [inline src/helpers/helpers.js:11]: I think the filter does not mutate the original Array so no need to destructure again

File: src/pages/Home/Home.js [hunk 5]
```diff
@@ -1,25 +1,49 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
```
Comment (kiranpratyush) [inline src/pages/Home/Home.js:1]: Two times React is imported

File: src/pages/Home/Home.js [hunk 6]
```diff
@@ -1,25 +1,49 @@
...
 import { NavSidebarContainer, VideoListing } from "../../components";
+import { useVideo } from "../../context/VideoContext";
```
Comment (kiranpratyush) [inline src/pages/Home/Home.js]: We can Name it as useVideoContext

**Review by kiranpratyush — APPROVED** [review]
UI looks, Cool bro. Nice Work

File: src/context/VideoContext.js (hunk 3 above)
Comment (AbhishekkGautam) [reply src/context/VideoContext.js:20]: it's declared. I missed dispatching the loading in the previous PR. so I'm just refactoring it here.

File: src/helpers/helpers.js (hunk 4 above)
Comment (AbhishekkGautam) [reply src/helpers/helpers.js:11]: yes. I did this for extra caution.

File: src/pages/Home/Home.js (hunk 5 above)
Comment (AbhishekkGautam) [reply src/pages/Home/Home.js:1]: No. It's only one time. the red one is from the previous merge. the green one is the updated one.

File: src/pages/Home/Home.js (hunk 6 above)
Comment (AbhishekkGautam) [reply src/pages/Home/Home.js]: done.

**Pull request merged**